src/common/cbor_values/tests.rs

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2022 The NAMIB Project Developers.
+ * Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+ * https://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+ * <LICENSE-MIT or https://opensource.org/licenses/MIT>, at your
+ * option. This file may not be copied, modified, or distributed
+ * except according to those terms.
+ *
+ * SPDX-License-Identifier: MIT OR Apache-2.0
+ */
+
+mod pop {
+    #[cfg(not(feature = "std"))]
+    use alloc::{string::String, string::ToString, vec};
+    use core::marker::PhantomData;
+
+    use ciborium::value::Value;
+    use coset::iana::Algorithm;
+    use coset::{
+        iana, CoseEncrypt0, CoseEncrypt0Builder, CoseKey, CoseKeyBuilder, HeaderBuilder,
+        ProtectedHeader,
+    };
+
+    use crate::common::test_helper::expect_ser_de;
+    use crate::error::WrongSourceTypeError;
+    use crate::ProofOfPossessionKey::{EncryptedCoseKey, KeyId, PlainCoseKey};
+    use crate::{ByteString, ProofOfPossessionKey, ToCborMap};
+
+    #[test]
+    fn test_key_id() -> Result<(), String> {
+        let pop = KeyId(vec![0xDC, 0xAF]);
+        assert_eq!(pop.key_id(), &vec![0xDC, 0xAF]);
+        assert_eq!(
+            ByteString::try_from(pop.clone()).expect("must be KeyId"),
```

Comment:
This error message seems confusing: Isn't the error that would cause this panic a conversion error between `KeyId` and `ByteString`?

The error message seems to be more fitting for a failure of the `assert_eq!` macro and not of the `ByteString::try_from`-call.

EDIT: After looking at other error messages in these tests, I noticed that the string should probably be "`must be ByteString`" to be consistent with other test cases.

```suggestion
            ByteString::try_from(pop.clone()).expect("must be ByteString"),
```

Reply:
Changed `ByteString` to `KeyId` in 03a129c00281e0d2939d289281d9e44f18d5d01e to make this more readable.